cmd/progressbar/progressbar.go

```diff
@@ -124,28 +130,40 @@ func (pb *progressBar) handleOptionalBarCreation(data OptionalBarCreation) {
 	)
 }
 
-func (pb *progressBar) handleBarUpdate(data BarUpdate) {
+func (pb *progressBar) handleDynamicBarUpdate(data DynamicBarUpdate) {
 	displayName := data.BarName
 
 	val, exists := pb.bars[displayName]
 	if !exists {
 		log.Panicf("trying to update a bar that doesn't exist: %s (%v)", displayName, data)
 	}
 
-	if data.Change <= 0 {
-		return
+	if data.TotalChange > 0 {
+		val.SetTotal(val.Current()+data.TotalChange, false)
 	}
 
-	if data.TotalChange > 0 {
-		val.SetTotal(val.Current()+int64(data.TotalChange), false)
+	if data.Change > 0 {
+		val.IncrBy(data.Change)
 	}
 
-	if data.TriggerDone {
+	if data.Close {
```

Comment:
remove data.Close from dynamic bar update,
add "ignoreUncompleted" to BarClosed and use it instead.